apps/selfhosted/pihole-cloudflared.yml

```diff
@@ -0,0 +1,69 @@
+---
+version: '3'
+services:
+  pihole:
+    depends_on:
+      - pi-cloudflared
+    image: ${PIHOLEIMAGE}
+    hostname: pihole
+    container_name: pihole
+    cap_add:
+      - NET_ADMIN
+    ports:
+      - '${PORTBLOCK}:53:53/tcp'
+      - '${PORTBLOCK}:53:53/udp'
+      - '${PORTBLOCK}:8008:80/tcp'
+    networks:
+      - proxy
+    dns:
+      - ${NS1}
+      - ${NS2}
+    environment:
+      - 'PGID=${ID}'
+      - 'PUID=${ID}'
+      - 'TZ=${TZ}'
+      - 'UMASK=${UMASK}'
+      - 'PIHOLE_DNS_=172.18.0.200#5053;172.18.0.200#5053'
```

Comment:
DNS_ ?

Reply:
@doob187 This uses either `cloudflared_DOH` or `unbound` as the upstream DNS server over the internal network.

Reply:
Example if you wanted to use simple Cloudflare & google here>
```yaml
      - 'PIHOLE_DNS_=1.1.1.1;8.8.8.8'
```

Reply:
@doob187 https://github.com/pi-hole/docker-pi-hole/#upgrade-notes